gears/src/types/base/coin.rs

```diff
@@ -18,6 +18,22 @@ pub struct Coin {
     pub amount: Uint256,
 }
 
+impl Coin {
+    pub fn amount_of<'a, I: Iterator<Item = &'a Coin>>(coins: I, denom: &Denom) -> Uint256 {
```

Comment:
What happens if the same denom is repeated? As above I think we should have a new type which captures the fact that there are no repeated denoms in a list. This function should be a method on that type.
There is no distinction between denom not found and finding the denom with an amount zero. This might be OK for how you use it but will likely cause problems in future. Better to be explicit and return an option?

Reply:
You right. I know about this restriction, and `.collect::<HashMap<_, _>>();` "resolves" this, but the question is what behavior should be.  Return error, panic or ignore as it does now

Reply:
Edit: see comment below